.github/workflows/multi-node-test.yml

```diff
@@ -0,0 +1,32 @@
+name: Integ tests with Multiple Nodes
+# This workflow is triggered on pull requests and pushes to main or an OpenSearch release branch
+on:
+  pull_request:
+    branches:
+      - "*"
+  push:
+    branches:
+      - "*"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        java: [ 21 ]
+    # Job name
+    name: Build and test Job-scheduler
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+    steps:
+      # This step uses the setup-java Github action: https://github.com/actions/setup-java
+      - name: Set Up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      # This step uses the checkout Github action: https://github.com/actions/checkout
+      - name: Checkout Branch
+        uses: actions/checkout@v4
+      - name: Run Job-scheduler Integ Tests with Multiple Nodes
+        run: |
+          echo "Running integ tests with multiple nodes..."
+          ./gradlew :opensearch-job-scheduler-sample-extension:integTest -PnumNodes=2
```

Comment:
LGTM, may be we need a matrix to run by default and run with `-PnumNodes=2`.

Reply:
ya that's a good idea. Some integTests in the root project run with multiple nodes and have their test suites annotated accordingly ([example](https://github.com/opensearch-project/job-scheduler/blob/c9857f0494eae08ec12540a39de52a24ced6e871/src/test/java/org/opensearch/jobscheduler/multinode/ReleaseLockActionMultiNodeRestIT.java#L19)).

The sample plugin is a useful tool for running integ tests with an actual plugin that schedules work so I think the integ tests can be expanded upon there esp as we add more functionality to the job scheduler.